iati_datastore/iatilib/crawler.py

```diff
@@ -294,7 +277,7 @@ def update_dataset(dataset_name, ignore_hashes):
     fetch_dataset_metadata(dataset)
     try:
         db.session.commit()
-    except sa.exc.IntegrityError as exc:
+    except (sa.exc.IntegrityError, sa.orm.exc.FlushError) as exc:
```

Comment:
There’s something funny going on here… I am not sure why this is necessary, but tests are throwing an error without it. I’ll look into this!